src/app/routes/cpsAsset/getInitialData/convertToOptimoBlocks/blocks/include/index.test.js

```diff
@@ -1,151 +1,192 @@
         required: false,
         tile: 'A quiz!',
         platform: 'highweb',
         type: 'idt1',
+        html: encodeHTML(idt1Markup),
+      },
+    };
+    expect(await convertInclude(input)).toEqual(expected);
+    expect(fetch).toHaveBeenCalled();
+  });
+
+  it('should convert an include block to an idt2 block with no / in href', async () => {
+    fetch.mockResponse(() => Promise.resolve(idt2Markup));
+    const input = {
+      required: false,
+      tile: 'IDT2 Include',
+      href: 'idt2',
+      platform: 'highweb',
+      type: 'include',
+    };
+    const expected = {
+      type: 'include',
+      model: {
+        href: 'idt2',
+        required: false,
+        tile: 'IDT2 Include',
+        platform: 'highweb',
+        type: 'idt2',
+        html: encodeHTML(idt2Markup),
       },
     };
-    expect(convertInclude(input)).toEqual(expected);
+    expect(await convertInclude(input)).toEqual(expected);
+    expect(fetch).toHaveBeenCalled();
   });
 
-  it('should convert an include block to an idt2 block with no / in href', () => {
+  it('should convert an include block to an idt2 block with html set to null when fetch returns with status other than 200', async () => {
+    jest.mock('#lib/logger.web', () => jest.fn());
+    fetch.mockResponse(() => Promise.resolve({ status: 304 }));
     const input = {
       required: false,
       tile: 'IDT2 Include',
       href: 'idt2',
       platform: 'highweb',
       type: 'include',
-      url: 'https://bbc.com/idt2.html',
     };
     const expected = {
       type: 'include',
       model: {
-        href: 'https://bbc.com/idt2.html',
+        href: 'idt2',
         required: false,
         tile: 'IDT2 Include',
         platform: 'highweb',
         type: 'idt2',
+        html: null,
       },
     };
-    expect(convertInclude(input)).toEqual(expected);
+    expect(await convertInclude(input)).toEqual(expected);
+    expect(fetch).toHaveBeenCalled();
   });
 
-  it('should return null for an unsupported include type', () => {
+  it('should return null for an unsupported include type', async () => {
     const input = {
       required: false,
       tile: 'A random include',
       href: '/idt3/111-222-333-444-555',
       platform: 'highweb',
       type: 'include',
-      url: 'https://bbc.com/idt20.html',
     };
-    expect(convertInclude(input)).toEqual(null);
+    expect(await convertInclude(input)).toEqual(null);
   });
 
-  it('should return null for an unsupported include type with no leading / in href', () => {
+  it('should return null for an unsupported include type with no leading / in href', async () => {
     const input = {
       required: false,
       tile: 'A random include',
       href: 'idt3/111-222-333-444-555',
       platform: 'highweb',
       type: 'include',
-      url: 'https://bbc.com/idt20.html',
     };
-    expect(convertInclude(input)).toEqual(null);
+    expect(await convertInclude(input)).toEqual(null);
```

Comment:
Should assert fetch hasn't been called here: `expect(fetch).not.toHaveBeenCalled();`

Reply:
it shouldn't. will remove it

Reply:
> it shouldn't. will remove it
I was suggesting adding the extra `expect(fetch).not.toHaveBeenCalled();` because for a non-supported include type no fetch will be made.